### PR TITLE
feat: Enhance UI, responsiveness, and fix deployment error

### PR DESCRIPTION
**Pull request by mutaremalcolm**
### Pull Request: Enhance UI, responsiveness, and fix deployment error

**Description:**
This pull request introduces significant enhancements aimed at improving the user interface, responsiveness, and deployment stability of the project. Here's a detailed breakdown of the changes:

- **UI Refactoring for Card Components:** The UI for card components has been refactored to improve visual aesthetics. Changes include adjustments to sizes and rendering styles, resulting in a more polished appearance.
  
- **Enhanced Responsiveness:** Various optimizations have been implemented to ensure optimal responsiveness across a wide range of devices and screen sizes. This ensures a consistent user experience regardless of the device used.
  
- **Fix Deployment Error:** Addressed an error that was causing build failures during deployment. There was an un-used component in the app.tsx which has been resolved.

**Details of Changes:**
- Refactored card component UI to improve visual appeal.
- Implemented responsive design enhancements for better adaptability across devices. //TODO: resolve bug on dessert and drinks sections
- Resolved a deployment error that was causing build failures.

These changes collectively elevate the quality and usability of the project, enhancing both its visual presentation and technical robustness.

### CODE REVIEW

File: test/flipDish.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { test, expect, render, screen } from "vitest";
```
Comment (DevonGifford) [inline test/flipDish.test.js:1]: PTAL:  On the folder structure 
The naming convention is to name the parent folder __tests__
This folder should also live inside your src folder 
Take a look at this project for reference (Vite + Vitest):  https://github.com/DevonGifford/ClearScore

File: vite.config.ts [hunk 2]
```diff
@@ -1,13 +1,25 @@
+/// <reference types="vitest" />
```
Comment (DevonGifford) [inline vite.config.ts:1]: LGTM

File: src/lib/apiDataFetch.tsx [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import { useEffect, useState } from "react";
+import axios from "axios";
```
Comment (DevonGifford) [inline src/lib/apiDataFetch.tsx:2]: NIT : If you're only using axios for its fetch method and not utilizing any of its additional features like its error handling, interceptors etc etc etc., then is this package really worth the overhead?  For these projects I try to stick to native things, until you no longer can or is worth.

File: src/App.tsx [hunk 4]
```diff
@@ -10,26 +10,23 @@ function App() {
...
-             bg-slate-50 hover:bg-slate-100"
-             key={sectionKey}>
-              <CardHeader className="flex gap-1 font-bold md:flex-col">
+      {menuData ? (
+        menuData.MenuSections.map((section) => {
+          const sectionKey = section.MenuSectionId;
+          
+          const sectionCard = section.MenuItems ? (
+            <Card
+              className="rounded-lg my-2 hover:scale-95 hover:bg-secondary/80 cursor-pointer bg-slate-50 hover:bg-slate-100"
```
Comment (DevonGifford) [inline src/App.tsx:19]: NIT:  I really dislike this hover effect

File: src/App.tsx [hunk 5]
```diff
@@ -10,26 +10,23 @@ function App() {
...
+              key={sectionKey}
+            >
+              <CardHeader className="flex gap-1 font-family: ui-serif font-bold md:flex-col">
                 <span className="text-blue-900 text-2xl sm:text-3xl md:text-4xl items-center capitalize tracking-wider pb-2">
                   {section.Name}
                 </span>
               </CardHeader>
-              <CardContent className="flex flex-col justify-end">
+              <CardContent className="flex flex-row justify-end">
```
Comment (DevonGifford) [inline src/App.tsx:27]: I think you are looking for some wrapping effect for when there are more than 3 or 4 cards, 
also some aligning and centering for when there are less than 1 or 2

File: src/App.css [hunk 6]
```diff
@@ -1,3 +1,80 @@
...
+    --destructive-foreground: 210 40% 98%;
+
+    --border: 214.3 31.8% 91.4%;
+    --input: 214.3 31.8% 91.4%;
+    --ring: 222.2 84% 4.9%;
+ 
+    --radius: 0.5rem;
+  }
+ 
+  .dark {
```
Comment (DevonGifford) [inline src/App.css:38]: NIT:  Are you using this dark globals css?  I assume its boilerplate but could be cut if not being used

File: @/Header.tsx
Comment (DevonGifford) [inline @/Header.tsx:1]: LGTM - nice move removing this stuff

**Review by DevonGifford — APPROVED** [review]
LGTM - approving PR.

Some nice QOL improvements here.
The commit history could be a little cleaner (naming & grouping)

Recommend squashing into single commit.  Leave to you to merge. 👍

File: test/flipDish.test.js (hunk 1 above)
Comment (mutaremalcolm) [reply test/flipDish.test.js:1]: Okay noted.

File: src/lib/apiDataFetch.tsx (hunk 3 above)
Comment (mutaremalcolm) [reply src/lib/apiDataFetch.tsx:2]: Okay understood, you mentioned this in our last code review will make the necessary changes.

File: src/App.css (hunk 6 above)
Comment (mutaremalcolm) [reply src/App.css:38]: Will get rid of this as well.

**Pull request merged**